File: C++/2021/1stOct_avyayjain.cpp
#include<iostream>
#include<math.h>
using namespace std;


bool prime(int n)
{
	// Corner case
	if (n <= 1)
		return false;
	for (int i = 2; i <= sqrt(n); i++)
		if (n % i == 0)
			return false;

	return true;
}

int fibonacci(int n)
{
	int a = 1, b = 1, c, i;
	if( n == 0)
		return 0;
    else if ( n == 1 )
    {
        cout << "1 ";
        return 0;
    }
    cout << "1 1 ";
	for(i = 3; i <= n; i++)
	{
        c = a + b;
        if ( c % 5 != 0 && !prime( c ))
        cout << c << " ";
        else
        cout << "0 ";
        a = b;
        b = c;
	}
	return b;
}

int main()
{
	int a = 12,b=27;
	fibonacci(a);  
	cout<<endl;
    fibonacci(b);
	return 0;
}

File: Python/2021/1stOct_avyayjain.py
def sort(l):
    n = l[-1]
    prime = [True for i in range(n + 1)]
    p = 2
    while (p * p <= n):
        if (prime[p] == True):
             
            for i in range(p ** 2, n + 1, p):
                prime[i] = False
        p += 1
    prime[0]= False
    prime[1]= False
    
    for i in range(len(l)):
        if prime[l[i]] or l[i]%5==0:
            l[i] = 0
    return l

def Fibonacci(n):
    l = []
    f1 = 0
    f2 = 1
    if (n < 1):
        return
    for x in range(1, n):
        l.append(f2)
        next = f1 + f2
        f1 = f2
        f2 = next
    return l
 
 

a = 12
b= 27
l = Fibonacci(a+1)
l = sort(l)
m= Fibonacci(b+1)
m= sort(m)
print(l)
print(m)
